pytorch_lightning/strategies/ddp.py

```diff
@@ -429,5 +429,5 @@ def teardown(self) -> None:
             # GPU teardown
             log.detail(f"{self.__class__.__name__}: moving model to CPU")
             self.lightning_module.cpu()
-            # clean up memory
-            torch.cuda.empty_cache()
+
+        self.accelerator.teardown()
```

Comment:
Should we add `self.accelerator.teardown()` to base class Strategy's `teardown` method as well?

Reply:
I think this PR is reproducing the logic. Add self.accelerator.teardown() could change teardown logic for some strategies. Maybe let's try that out in a separate PR?

Reply:
> Should we add self.accelerator.teardown() to base class Strategy's teardown method as well?

Adding it to the base makes sense to me. However, we need to be aware of the `super().teardown()` call. 
We have the lines  `self.lightning_module.cpu()` so the call to empty cuda cache should run after that.

I'm good with the current version. If you change the PR to what @kaushikb11 suggested, please re-request review. Otherwise LGTM

Reply:
Let's keep the PR atomic. One PR for replacing the code, one PR to move this in the base class.